src/VisualStudio/CSharp/Impl/ProjectSystemShim/CSharpProjectShim.cs

```diff
@@ -302,12 +274,10 @@ protected CSharpParseOptions CreateParseOptions()
             var languageVersion = CompilationOptionsConversion.GetLanguageVersion(GetStringOption(CompilerOptions.OPTID_COMPATIBILITY, defaultValue: ""))
                                   ?? CSharpParseOptions.Default.LanguageVersion;
 
-            return new CSharpParseOptions(
-                kind: SourceCodeKind.Regular,
-                languageVersion: languageVersion,
-                preprocessorSymbols: symbols.AsImmutable(),
-                documentationMode: documentationMode)
-                .WithFeatures(GetParsedCommandLineArguments().ParseOptions.Features);
+            return options.WithKind(SourceCodeKind.Regular)
```

Comment:
I think that holds if we are creating options from scratch. As the default workspace options match for current AbstractLegacyProject and CPS based projects, I have moved that code down to the AbstractProject.GetCompilationOptions(), which necessitates the sub-types to modify the base options with .With methods.